src/apple-utils.js

```diff
@@ -1,32 +1,4 @@
-function getTopLevelURL () {
-    try {
-        // FROM: https://stackoverflow.com/a/7739035/73479
-        // FIX: Better capturing of top level URL so that trackers in embedded documents are not considered first party
-        if (window.location !== window.parent.location) {
-            return new URL(window.location.href !== 'about:blank' ? document.referrer : window.parent.location.href)
-        } else {
-            return new URL(window.location.href)
-        }
-    } catch (error) {
-        return new URL(location.href)
-    }
-}
-
-function isUnprotectedDomain (topLevelUrl, featureList) {
-    let unprotectedDomain = false
-    const domainParts = topLevelUrl && topLevelUrl.host ? topLevelUrl.host.split('.') : []
-
-    // walk up the domain to see if it's unprotected
-    while (domainParts.length > 1 && !unprotectedDomain) {
-        const partialDomain = domainParts.join('.')
-
-        unprotectedDomain = featureList.filter(domain => domain.domain === partialDomain).length > 0
-
-        domainParts.shift()
-    }
-
-    return unprotectedDomain
-}
+import { getTopLevelURL, isUnprotectedDomain } from './utils'
 
 export function processConfig (data, userList, preferences) {
```

Comment:
Thanks for sharing the common methods above :)

Does it make sense to share this with windows also; it seems like there's very little difference. Or break it into even smaller functions perhaps?

Reply:
Yeah, you're right I moved that one to the common utils as well.